core/Azure.Mcp.Core/tests/Azure.Mcp.Core.UnitTests/Areas/Server/ServiceStartCommandTests.cs

```diff
@@ -458,6 +529,26 @@ private ParseResult CreateParseResultWithAllOptions()
         return _command.GetCommand().Parse([.. args]);
     }
 
+    private ParseResult CreateParseResultWithTool(string[]? tools)
+    {
+        var args = new List<string>
+        {
+            "--transport", "stdio",
+            "--mode", "all"
```

Comment:
If the new --tool flag can be used together with the namespace mode, could you add a few more test cases to validate cases where this --mode is "namespace"?

Reply:
There was some discussion happened offline. We agreed to automatically switch to `all` mode when `--tool` is used.